### PR TITLE
Migrated to the new azidentity library

### PR DESCRIPTION
**Pull request by gapra-msft**
azidentity library used for SPN and MSI

custom TokenCredential created for device code and token store

### CODE REVIEW

**Review by tasherif-msft — APPROVED** [review]

**Review by nakulkar-msft — APPROVED** [review]

File: common/oauthTokenManager.go [hunk 1]
```diff
@@ -148,309 +131,92 @@ func (uotm *UserOAuthTokenManager) GetTokenInfo(ctx context.Context) (*OAuthToke
...
+	oAuthTokenInfo := &OAuthTokenInfo{
+		Identity:     true,
+		IdentityInfo: identityInfo,
 	}
 
-	return &oAuthTokenInfo, nil
+	return uotm.validateAndPersistLogin(oAuthTokenInfo, persist)
 }
 
 // SecretLogin is a UOTM shell for secretLoginNoUOTM.
```
Comment (adreed-msft) [inline common/oauthTokenManager.go:177]: This comment isn't entirely true anymore.

File: common/oauthTokenManager.go [hunk 2]
```diff
@@ -627,7 +394,7 @@ const TokenRefreshSourceTokenStore = "tokenstore"
 
 // OAuthTokenInfo contains info necessary for refresh OAuth credentials.
 type OAuthTokenInfo struct {
-	azcore.TokenCredential
+	azcore.TokenCredential `json:"-"`
```
Comment (adreed-msft) [inline common/oauthTokenManager.go:397]: So the current flow for login is `azcopy login` -> `azcopy whatever` (outside of one-shot login). If we're not persisting the refresh info between runs, then interactive login is completely broken. Are we saving that info some other way?

File: common/oauthTokenManager.go [hunk 3]
```diff
@@ -627,7 +394,7 @@ const TokenRefreshSourceTokenStore = "tokenstore"
...
 	adal.Token
```
Comment (adreed-msft) [inline common/oauthTokenManager.go:398]: What is the purpose of maintaining two tokens?

**Review by adreed-msft — APPROVED** [review]

File: common/oauthTokenManager.go (hunk 3 above)
Comment (gapra-msft) [reply common/oauthTokenManager.go:398]: the adal.Token needs to be used until we can entirely get off of Track 1.5 and it's sole purpose is to hold the refresh token for device code cred

File: common/oauthTokenManager.go (hunk 1 above)
Comment (gapra-msft) [reply common/oauthTokenManager.go:177]: Fixed the comment.

File: common/oauthTokenManager.go (hunk 2 above)
Comment (gapra-msft) [reply common/oauthTokenManager.go:397]: I took a look at the code again and we should be persisting the token between runs. We can discuss the code changes further if you'd like.

**Pull request merged**